### ISSUE TITLE
Using EleutherAI/enformer-official-rough PyTorch implementation to just get human output head

### ISSUE THREAD

**Issue by aaronwtr** [lucidrains/enformer-pytorch#15]
Hi @lucidrains,

Thank you so much for your efforts in releasing the PyTorch version of the Enformer model! I am really excited to use it for my particular implementation.

I was wondering if it is possible to use the pre-trained huggingface model to just get the human output head. The reason is that inference takes a few minutes, and since I just need human data, this will help make my implementation a bit smoother. Is there a way to do this elegantly with the current codebase, or would I need to rewrite some functions to allow for this? From what I have seen so far it doesn't seem that this modularity is possible yet. 

The way I have set up my inference currently is as follows: 
```python
class EnformerInference:
    def __init__(self, data_path: str, model_path="EleutherAI/enformer-official-rough"):
        if torch.cuda.is_available():
            device = torch.device("cuda")
        else:
            device = torch.device("cpu")
        self.device = device
        self.model = Enformer.from_pretrained(model_path)
        self.data = EnformerDataLoader(pd.read_csv(data_path, sep="\t")) # returns a one hot encoded torch.Tensor representation of the sequence of interest
                                                                                                                          

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        return self.model(x.to(self.device))
```

Any guidance on this would be greatly appreciated, thank you!

**Comment by lucidrains** [comment]
Hey there!

So actually you won't save that much even if you only return the human output head, since the bulk of the processing happens in the transformer layers preceding the linear heads

How many sequences are you fetching the embeddings for? A few minutes sounds too long for a single sequence, unless if you are somehow doing this on a cpu, which I do not recommend

Phil

**Comment by aaronwtr** [comment]
Hi, 

Thanks! You were right in that I was using CPU. I was fetching embeddings for three sequences. I tried to switch to my mac's GPU (new mac user here) but so far unsuccessful. I am using the new MPS backend that is supposed to support Apple's new M1/M2 chips but I am getting the following error: 

`TypeError: Operation 'abs_out_mps()' does not support input type 'int64' in MPS backend.`

I have double checked that all the input tensors are casted to float32 torch tensors, but I have found that some of Enformer's weights are casted as int64, which I think is causing the issue. I haven't been successful in recasting the model weights to torch.float32, I tried the following: 

```python
weights = self.model.state_dict()
for name, param in weights.items():
    weights[name] = param.to(torch.float32)
self.model = self.model.load_state_dict(weights)
```
but that does not preserve self.model to be a pretrained model object as we get from ` self.model = Enformer.from_pretrained(model_path)`. 

If you have any suggestions on this I'd greatly appreciate it. Otherwise I think I'll have to run inference on CPU until PyTorch compatibility with M1/M2 is fixed.
